src/bin/lib/pg/snprintf.c

```diff
@@ -31,6 +31,8 @@
  * src/port/snprintf.c
  */
 
+#ifndef USE_REPL_SNPRINTF
```

Comment:
This needs to be done after including the postgres header that sets this, either `port.h` (PG12) or `confdefs.h` (PG11/PG10). It could be these are both included in `postgres_fe.h` and/or `c.h`, so doing this check after including those might be easier. The way this check is done now I would expect it to always be true. Same for the cases below. 

One way to confirm that the check is working is to put the `#error "should not be reached"` in it and confirm that it still compiles on PG12.

Reply:
Thanks for the careful review, done that way now. I can confirm the `#error` verification on a manual build test here.